dbt/adapters/athena/session.py

```diff
@@ -102,46 +108,16 @@ def athena_client(self) -> Any:
             "athena", config=get_boto3_config(num_retries=self.credentials.effective_num_retries)
         )
 
-    def get_new_sessions(self) -> List[UUID]:
+    @cached_property
+    def session_description(self) -> str:
```

Comment:
question: Should this be a cached property? The session manager will be handling multiple sessions and as such a single session is not a property of the `AthenaSparkSessionManager`

Reply:
it is just engine config and invocation id which distinguishes a session. So for a given session this will be its property which i am using to tag the sessions while creating them.
And i tested this out by running 11 models.
3 models with same configuration shared the same session and others split into different sessions based on their engine config.

![image](https://github.com/Avinash-1394/dbt-athena-1/assets/135387788/c84f7ba6-45d9-4567-97e0-60c77c4ce215)

Below are the sample models i used for testing this session setup:
[models.zip](https://github.com/Avinash-1394/dbt-athena-1/files/13816049/models.zip)

Reply:
Got it! That makes sense! 

Do you mind if I add those models to https://github.com/Avinash-1394/dbt-spark-development? It would be good to provide that repo as a reference to people new to athena spark.

Reply:
Please use them but they are all same models and same config. The different one I added to readme.